<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SearchWidget</class>
 <widget class="QWidget" name="SearchWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>311</width>
    <height>454</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QFormLayout" name="formLayout">
   <property name="fieldGrowthPolicy">
    <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
   </property>
   <item row="0" column="0">
    <widget class="QLabel" name="label">
     <property name="font">
      <font>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="text">
      <string>Search:</string>
     </property>
    </widget>
   </item>
   <item row="0" column="1">
    <widget class="QLineEdit" name="lineEdit">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
       <horstretch>1</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="toolTip">
      <string>Search the encrypted database</string>
     </property>
     <property name="whatsThis">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:10pt;&quot;&gt;The search string can either be a wildcard or regular expression, depending on the selection below&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
    </widget>
   </item>
   <item row="2" column="0" colspan="2">
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Options</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="0">
       <widget class="QCheckBox" name="chk_caseSensitive">
        <property name="toolTip">
         <string>Should search be case-sensitive?</string>
        </property>
        <property name="whatsThis">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:10pt;&quot;&gt;Controls whether the search string should be interpreted as case-sensitive&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="text">
         <string>Case Sensitive</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QCheckBox" name="chk_onlyFiles">
        <property name="toolTip">
         <string>Show only entries with files in the view</string>
        </property>
        <property name="text">
         <string>Show Only  Files</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QCheckBox" name="chk_onlyFavorites">
        <property name="toolTip">
         <string>Show only favorites in the view</string>
        </property>
        <property name="text">
         <string>Show Only Favorites</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QCheckBox" name="chk_filter_results">
        <property name="toolTip">
         <string>Should the view filter the unmatched entries?</string>
        </property>
        <property name="whatsThis">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:10pt;&quot;&gt;If this is checked, then any rows that don't match the search string will be hidden from view. Note that entries whose ancestors match the search string are still shown.&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="text">
         <string>Filter Results</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QCheckBox" name="chk_alsoSecrets">
        <property name="toolTip">
         <string>Search the secret values, such as passwords</string>
        </property>
        <property name="whatsThis">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:10pt;&quot;&gt;Normally you don't want to search your passwords, but in case one of your passwords gets compromised you can use this function to find every entry where your password is used so you can update them.&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="text">
         <string>Also Search Secrets</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="5" column="0" colspan="2">
    <widget class="QPushButton" name="btn_clear">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>1</verstretch>
      </sizepolicy>
     </property>
     <property name="whatsThis">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:10pt;&quot;&gt;Clears the search, re-showing all entries in the view&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
     <property name="text">
      <string>Clear Filter</string>
     </property>
    </widget>
   </item>
   <item row="3" column="0" colspan="2">
    <widget class="QGroupBox" name="gb_filterType">
     <property name="title">
      <string>Search String Type</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="1" column="0">
       <widget class="QRadioButton" name="rdo_regexp">
        <property name="whatsThis">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:10pt;&quot;&gt;If this is checked, then the search string is interpreted as a regular expression. Regexp's are much more powerful than wildcards, but they are more difficult to use for novice users.&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="text">
         <string>Regular Expression</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QRadioButton" name="rdo_wildCard">
        <property name="whatsThis">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:10pt;&quot;&gt;If this is checked, then the search string is interpreted as a wildcard expression. That means you can use the asterisk to match any amount of any characters, and a question mark to match any single character.&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="text">
         <string>Wildcard</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="4" column="0" colspan="2">
    <widget class="QGroupBox" name="gb_time">
     <property name="whatsThis">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;&lt;span style=&quot; font-size:10pt;&quot;&gt;Select a time range by which to filter entries&lt;/span&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
     <property name="title">
      <string>Time Range Selection</string>
     </property>
     <property name="flat">
      <bool>false</bool>
     </property>
     <property name="checkable">
      <bool>true</bool>
     </property>
     <property name="checked">
      <bool>false</bool>
     </property>
     <layout class="QFormLayout" name="formLayout_2">
      <property name="fieldGrowthPolicy">
       <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
      </property>
      <item row="0" column="0">
       <widget class="QCheckBox" name="chk_start">
        <property name="text">
         <string>Start Time</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QDateTimeEdit" name="dte_start"/>
      </item>
      <item row="1" column="0">
       <widget class="QCheckBox" name="chk_end">
        <property name="text">
         <string>End Time</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QDateTimeEdit" name="dte_end"/>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>lineEdit</sender>
   <signal>textChanged(QString)</signal>
   <receiver>SearchWidget</receiver>
   <slot>_something_changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>121</x>
     <y>24</y>
    </hint>
    <hint type="destinationlabel">
     <x>181</x>
     <y>34</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rdo_wildCard</sender>
   <signal>clicked()</signal>
   <receiver>SearchWidget</receiver>
   <slot>_something_changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>120</x>
     <y>270</y>
    </hint>
    <hint type="destinationlabel">
     <x>2</x>
     <y>107</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rdo_regexp</sender>
   <signal>clicked()</signal>
   <receiver>SearchWidget</receiver>
   <slot>_something_changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>85</x>
     <y>298</y>
    </hint>
    <hint type="destinationlabel">
     <x>2</x>
     <y>136</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>gb_time</sender>
   <signal>clicked()</signal>
   <receiver>SearchWidget</receiver>
   <slot>_something_changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>28</x>
     <y>411</y>
    </hint>
    <hint type="destinationlabel">
     <x>0</x>
     <y>171</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>chk_start</sender>
   <signal>clicked()</signal>
   <receiver>SearchWidget</receiver>
   <slot>_something_changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>98</x>
     <y>362</y>
    </hint>
    <hint type="destinationlabel">
     <x>129</x>
     <y>157</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>chk_end</sender>
   <signal>clicked()</signal>
   <receiver>SearchWidget</receiver>
   <slot>_something_changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>72</x>
     <y>395</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>158</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>dte_start</sender>
   <signal>dateTimeChanged(QDateTime)</signal>
   <receiver>SearchWidget</receiver>
   <slot>_something_changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>292</x>
     <y>367</y>
    </hint>
    <hint type="destinationlabel">
     <x>394</x>
     <y>184</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>dte_end</sender>
   <signal>dateTimeChanged(QDateTime)</signal>
   <receiver>SearchWidget</receiver>
   <slot>_something_changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>292</x>
     <y>400</y>
    </hint>
    <hint type="destinationlabel">
     <x>394</x>
     <y>232</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>chk_caseSensitive</sender>
   <signal>stateChanged(int)</signal>
   <receiver>SearchWidget</receiver>
   <slot>_something_changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>292</x>
     <y>94</y>
    </hint>
    <hint type="destinationlabel">
     <x>344</x>
     <y>35</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btn_clear</sender>
   <signal>clicked()</signal>
   <receiver>SearchWidget</receiver>
   <slot>Clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>166</x>
     <y>444</y>
    </hint>
    <hint type="destinationlabel">
     <x>218</x>
     <y>285</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>chk_filter_results</sender>
   <signal>stateChanged(int)</signal>
   <receiver>SearchWidget</receiver>
   <slot>_something_changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>80</x>
     <y>122</y>
    </hint>
    <hint type="destinationlabel">
     <x>111</x>
     <y>45</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>chk_onlyFavorites</sender>
   <signal>clicked()</signal>
   <receiver>SearchWidget</receiver>
   <slot>_something_changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>148</x>
     <y>133</y>
    </hint>
    <hint type="destinationlabel">
     <x>165</x>
     <y>220</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>chk_onlyFiles</sender>
   <signal>clicked()</signal>
   <receiver>SearchWidget</receiver>
   <slot>_something_changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>113</x>
     <y>172</y>
    </hint>
    <hint type="destinationlabel">
     <x>106</x>
     <y>219</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>chk_alsoSecrets</sender>
   <signal>clicked()</signal>
   <receiver>SearchWidget</receiver>
   <slot>_something_changed()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>135</x>
     <y>200</y>
    </hint>
    <hint type="destinationlabel">
     <x>4</x>
     <y>245</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>_something_changed()</slot>
  <slot>Clear()</slot>
 </slots>
</ui>
